### PR TITLE
[MPLUGINTESTING-94] Switch to Maven 4.0.0-beta-3

### PR DESCRIPTION
**Pull request by gnodet**
This PR switches `maven-plugin-testing` to support plugins based on the Maven 4 API.
Support for maven 3 plugins is removed, as there is no code sharing at all, so it's easier to split both code bases imho.

### CODE REVIEW

File: maven-plugin-testing-harness/pom.xml [hunk 1]
```diff
@@ -74,6 +80,15 @@ under the License.
       <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven</groupId>
+      <artifactId>maven-xml-impl</artifactId>
+      <version>${mavenVersion}</version>
```
Comment (slawekjaranowski) [inline maven-plugin-testing-harness/pom.xml:86]: Should be optional or provided to allow use with 3.x

**Review by slawekjaranowski — COMMENTED** [review]
Please use Maven 4 artifats in provided scope or optional to allow use library with Maven 3.x also.

File: maven-plugin-testing-harness/pom.xml (hunk 1 above)
Comment (gnodet) [reply maven-plugin-testing-harness/pom.xml:86]: I've removed the dependency which is not actually used here.  I'll send an email to the dev list about moving some modules around.

**Comment by gnodet** [comment]
@slawekjaranowski @cstamas could you have a look at this one ?

**Comment by cstamas** [comment]
LGTM

**Pull request merged**